code/modules/unit_tests/say.dm

```diff
@@ -156,5 +168,37 @@
 	handle_speech_result = null
 	handle_hearing_result = null
 
+/datum/unit_test/speech/proc/radio_test()
+	speaker.forceMove(run_loc_floor_bottom_left)
+	listener.forceMove(locate((run_loc_floor_bottom_left.x + 10), run_loc_floor_bottom_left.y, run_loc_floor_bottom_left.z))
+
+	speaker_radio.forceMove(run_loc_floor_bottom_left)
+	speaker_radio.set_broadcasting(TRUE)
+	listener_radio.forceMove(locate((run_loc_floor_bottom_left.x + 10), run_loc_floor_bottom_left.y, run_loc_floor_bottom_left.z))
+	// Normally speaking, if there isn't a functional telecomms array on the same z-level, then handheld radios
+	// have a short delay before sending the message. We use the centcom frequency to get around this.
+	speaker_radio.set_frequency(FREQ_CENTCOM)
+	speaker_radio.independent = TRUE
+	listener_radio.set_frequency(FREQ_CENTCOM)
+	listener_radio.independent = TRUE
+
+	var/pangram_quote = "The quick brown fox jumps over the lazy dog"
+
+	speaker.say(pangram_quote)
+	TEST_ASSERT(handle_speech_result, "Handle speech signal was not fired (radio test)")
+	TEST_ASSERT_EQUAL(islist(handle_hearing_result), speaker_radio.get_frequency() == listener_radio.get_frequency(), "Listener failed to hear radio message (radio test)")
```

Comment:
this is not how `TEST_ASSERT_EQUAL()` should be used imo, feels incredibly ugly and unworth. do it like this:

```suggestion
	TEST_ASSERT(islist(handle_hearing_result), "error message here")
	TEST_ASSERT_EQUAL(speaker_radio.get_frequency(), listener_radio.get_frequency(), "Listener failed to hear radio message (radio test)")
```

otherwise the syntax is just really confusing and unworth